core/include/traccc/clusterization/clusterization_algorithm.hpp

```diff
@@ -41,7 +43,9 @@ class clusterization_algorithm
     ///
     /// @param mr The memory resource to use for the result objects
     ///
-    clusterization_algorithm(vecmem::memory_resource& mr);
+    clusterization_algorithm(
+        vecmem::memory_resource& mr,
+        std::unique_ptr<const Logger> logger = getDummyLogger().clone());
```

Comment:
:confused: Technically clients don't need to change their code when instantiating these classes. They'll "just" get absolutely meaningless printouts if they don't.

This is what I don't understand. As long as we use default arguments here, why not make them such that in "standalone mode" we wouldn't even need to override them? Only once we integrate the tools into a larger framework. (...cough... Athena ...cough...)

Reply:
I'm not sure I really understand your argument, but I think your point is that you want users to be able to get meaningful logging even with just default parameters? Afaik that is only possible by having some kind of global state, and I absolutely want to avoid that.